pycaret/time_series.py

```diff
@@ -2008,31 +2027,409 @@ def get_folds(cv, y) -> Tuple[pd.Series, pd.Series]:
         yield rolling_y_train.index, y_test.index
 
 
-def cross_validate_ts(forecaster, y, X, cv, scoring, fit_params, n_jobs, return_train_score, error_score=0):
-    from pycaret.time_series import get_folds
+def cross_validate_ts(forecaster, y, X, scoring, cv, n_jobs, verbose, fit_params, return_train_score, error_score=0):
+    """Performs Cross Validation on time series data
+
+    Parallelization is based on `sklearn` cross_validate function [1]
+    Ref:
+    [1] https://github.com/scikit-learn/scikit-learn/blob/0.24.1/sklearn/model_selection/_validation.py#L246
+
+
+    Parameters
+    ----------
+    forecaster : [type]
+        [description]
+    y : [type]
+        [description]
+    X : [type]
+        [description]
+    cv : [type]
+        [description]
+    scoring : [type]
+        [description]
+    fit_params : [type]
+        [description]
+    n_jobs : [type]
+        [description]
+    return_train_score : [type]
+        [description]
+    error_score : int, optional
+        [description], by default 0
+
+    Returns
+    -------
+    [type]
+        [description]
+
+    Raises
+    ------
+    ValueError
+        [description]
+    """
+    try:
+        # # For Debug
+        # n_jobs = 1
+
+        parallel = Parallel(n_jobs=n_jobs)
+        out = parallel(delayed(_fit_and_score)(
+            forecaster=clone(forecaster),
+            y=y, X=X,
+            scoring=scoring,
+            train=train,
+            test=test,
+            parameters=None,
+            fit_params=fit_params,
+            return_train_score=return_train_score,
+            error_score=error_score)
+            for train, test in get_folds(cv, y))
+    # raise key exceptions
+    except (TerminatedWorkerError, KeyboardInterrupt, SystemExit):
```

Comment:
I'm just wondering if an Out of Memory Error could also happen ?

Reply:
I guess it could. We can add them as we encounter them. What do you think?